### DISCUSSION TITLE
Using the Spring Boot starter, otel expects to be autoconfigured but nothing happens

### DISCUSSION THREAD

**Discussion by rpgdev** [open-telemetry/opentelemetry-java-instrumentation#11765] [Q&A]
I've been able to instrument my code using the agent and autoconfiguration but using the SB starter doesn't seem to work nor error out. I get the following in the console and that's about it:
`INFO 291666 --- [           main] i.opentelemetry.api.GlobalOpenTelemetry  : AutoConfiguredOpenTelemetrySdk found on classpath but automatic configuration is disabled. To enable, run your JVM with -Dotel.java.global-autoconfigure.enabled=true`

But when I add `-Dotel.java.global-autoconfigure.enabled=true` it doesn't pick any of the properties from the properties file (as I expect it to do while using the SB starter). Not sure if that's the expectation. What could be the reason for this and how could I fix it?

**Comment by laurit** [comment]
cc @zeitlinger

**Comment by zeitlinger** [comment]
are you using this guide https://opentelemetry.io/docs/zero-code/java/spring-boot-starter/getting-started/ ?

**Comment by rpgdev** [comment]
@zeitlinger yep, exactly that plus the dependencies below. It's a propietary sb service so can't post the code, sorry. Other than that message there's nothing else from otel, how can I get a more verbose output? 
```xml
<dependency>
    <groupId>io.opentelemetry</groupId>
    <artifactId>opentelemetry-api</artifactId>
</dependency>
<dependency>
    <groupId>io.opentelemetry</groupId>
    <artifactId>opentelemetry-exporter-otlp</artifactId>
</dependency>
```

**Comment by zeitlinger** [comment]
- please add `debug=true` to your spring properties and look for beans with OpenTelemetry in it
- if all looks good, try the [console exporter](https://opentelemetry.io/docs/languages/java/configuration/#logging-exporter) to see if maybe the problem is not in the app
- if you still don't have the problem, create a stripped down version of your app without the proprietary parts

**Comment by rpgdev** [comment]
@zeitlinger in the end it was not necessary to do any of that once the starter reached GA status. I was able to use it by adding the dependencies as the guide says and it worked straight away. Probably the `alpha` dependency was conflicting with something.. anyways, thanks!

**Reply by yatamraghu7** [reply]
Hi @zeitlinger 

I am also facing the same problem but in our project I am only using below dependnecies

```
  <dependency>
            <groupId>io.opentelemetry.instrumentation</groupId>
            <artifactId>opentelemetry-spring-boot-starter</artifactId>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-extension-trace-propagators</artifactId>
        </dependency>
```

I have added the property to my applciation properties and it is not picking as well. Can you tell what you did in brief a little so that I can try that as well.

Thanks

**Reply by zeitlinger** [reply]
> I am also facing the same problem but in our project I am only using below dependnecies

can you provide a reproducible example?

**Reply by yatamraghu7** [reply]
Hi @zeitlinger 

Sorry that I wont be able to reproduce the error with some sample application as I dint understand why I am getting this error in the first place in our application. If I understand then may be I can provide a sample code snippet or small app which reproduces the issue.

Thanks

**Reply by zeitlinger** [reply]
can you try to take your app that is failing and reduce it so that you can share it?
Otherwise I can't diagnose the issue.

**Reply by yatamraghu7** [reply]
Hi @zeitlinger 

please find the sample application which reproduce the error here

https://github.com/yatamraghu7/ManualInstrumentationOtel

Also I am getting 00000000000000000 in my traceId and this is because of no proper tracer found i think at that point could that be because of this issue ?

Thanks

**Reply by yatamraghu7** [reply]
Hi @zeitlinger 

Did you get a chance to look at the code that was provided ?

Thanks

**Reply by zeitlinger** [reply]
- does the problem only occur with B3 propagation?
- please use the latest release https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v2.8.0 - and don't mix versions (you can use the alpha bom to have versions)
- opentelemetry-spring-webflux-5.3 is included in the stater already - shouldn't need to be added manually - same question for https://github.com/yatamraghu7/ManualInstrumentationOtel/blob/be804eee9dd12f6ea441a324325f11bfc563fac6/Consumer/src/main/java/com/microservice/consumer/Consumer/Config/WebClientConfig.java#L14-L32 
- what are the steps to get "Also I am getting 00000000000000000"?

**Reply by yatamraghu7** [reply]
Hi @zeitlinger 

I did opened a new discussion here https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/11765
can you please check once.

THanks
